src/runner.rs

```diff
@@ -217,6 +220,7 @@ fn render_data_test(desc: &DataTestDesc, rendered: &mut Vec<TestDescAndFn>) {
                 ignore: desc.ignore,
                 should_panic: ShouldPanic::No,
                 allow_fail: false,
+                test_type: TestType::IntegrationTest,
```

Comment:
Can these be made to work the same way as built-in `#[test]` framework? I haven't looked how rustc does it, but would be nice to emulate standard behavior.

Reply:
What do you mean? The choice of TestType?

Reply:
Right. Choice of TestType (UnitTest vs IntegrationTest).

Reply:
So this is how it's done, I think: https://github.com/popzxc/rust/blob/15f571bbd3ce370d925d21f1ab281aee993e4860/src/libsyntax_ext/test.rs#L286-L306

Is it even possible to have datatest in `src/`? If not, then it's always `IntegrationTest`.

Reply:
(And even then -- not sure how to go about checking this outside libsyntax.)

Reply:
Done. For posterity the single effect of this matching (as of now) is that the default timing limits for warn/critical markings, and failures with `--ensure-test-time`, will have different defaults for integration tests (`tests/*`) than for unit tests (`src/*`). I still don't know if that's relevant to datatest, but whatever, it works!